sbi/inference/trainers/nle/mnle.py

```diff
@@ -162,6 +162,20 @@ def build_posterior(
             Posterior $p(\theta|x)$  with `.sample()` and `.log_prob()` methods
             (the returned log-probability is unnormalized).
         """
+
+        warn_if_deprecated(
+            self.build_posterior,
+            locals(),
+            {
+                "mcmc_parameters",
+                "vi_parameters",
+                "rejection_sampling_parameters",
+                "importance_sampling_parameters",
+                "mcmc_method",
+                "vi_method",
+            },
```

Comment:
please use explicit kwargs here and all other calls to make the code more readable. e.g., 
```python
     warn_if_deprecated(
            method_to_check=self.build_posterior,
            local_args=locals(),
            deprecated_keys={
                "mcmc_parameters",
                "vi_parameters",
                "rejection_sampling_parameters",
                "importance_sampling_parameters",
                "mcmc_method",
                "vi_method",
            },
     )
```